torchaudio/csrc/ffmpeg/buffer.cpp

```diff
@@ -0,0 +1,163 @@
+#include <torchaudio/csrc/ffmpeg/buffer.h>
+#include <stdexcept>
+#include <vector>
+
+namespace torchaudio {
+namespace ffmpeg {
+
+Buffer::Buffer(AVMediaType type) : media_type(type) {}
+
+namespace {
+torch::Tensor convert_audio_tensor(AVFrame* pFrame) {
+  // ref: https://ffmpeg.org/doxygen/4.1/filter__audio_8c_source.html#l00215
+  AVSampleFormat format = static_cast<AVSampleFormat>(pFrame->format);
+  int num_channels = pFrame->channels;
```

Comment:
I feel like ``num_channels`` can be defined when initializing the Buffer object? As each Buffer object will receive signal from one Streamer, which fixed the num_channels after initialization.

Reply:
I agree with you, but unfortunately the shape information is not necessarily available at the time Buffer object is initialized, due to a filter graph operation inserted before the buffer, which can change number of audio channels, image size, image channels and rate. The structure passed around does not have the explicit information of these.

Let me figure this out later. along with the buffer size concern you brought up.